index.bs

```diff
@@ -253,18 +253,251 @@ Control Protocols {#control}
+
+- connection-id: The ID from the presentation-start-response or
+presentation-connection-open-response messages.
+
+- message: the presentation message data.  
+
+
+To terminate a presentation, the controller may send a
+presentation-termination-request message with the following values:
+
+- presentation-id: The ID of the presentation to terminate.
+
+- reason: The reason the presentation is being terminated.
+
+
+When a receiving endpoint receives a presentation-termination-request, it should
+send back a presentation-termination-response message to the requesting
+endpoint.  It may also notify other controllers about the termination by sending
+a presentation-termination-event message.  And it can send the same message if
+it terminates a presentation without a request from a controller to do so. This
+message contains the following values:
+
+- presentation-id: The ID of the presentation that was terminated.
+
+- reason: The reason the presentation was terminated.
+
+<!-- TODO: Split up reason into reason and whether it was triggered by the user
+or not? -->
+
+
+To accept incoming connections requests from controller, a receiver
+must receive and process the presentation-connection-open-request
+message which contains the following values: 
+
+- presentation-id: The ID of the presentation to connect to.
+
+
+The receiver should, upon receipt of a
+presentation-connection-open-request message, send back a
+presentation-connection-open-response message which contains the
+following values:
+
+- result: a code indicating success or failure, and the reason for the failure
+
+- connection-id: An ID that both endpoints can use to send connection messages
+to each other.  It is chosen by the receiver for ease of implementation (if the
+message receiver chooses the connection-id, it may keep the ID unique across
+connections, thus making message demuxing/routing easier).
+
+
+A controller may terminate a connection without terminating the presentation by
+sending a presentation-connection-close-request message with the following
+values:
+
+- connection-id: The ID of the connection to close.
+
+
+The receiver should, upon receipt of a presentation-connection-close-request,
+send back a presentation-connection-close-response message with the following
+values:
+
+- result: If the close succeed or failed, and if it failed why it failed.
+
+The receiver may also close a connection without a request from the controller
+to do so and without terminating a presentation.  If it does so, it should send
+a presentation-connection-close-event to the controller with the following
+values:
+
+- connection-id: The ID of the connection that was closed
+
+- reason: The reason the connection was closed
+
+- error-message: A debug message suitable for a log or perhaps presented to the
+user with more explanation as to why it was closed.
+
+<!-- TODO: Why does the Presentation API spec not mention the use of the close
+message? -->
```

Comment:
If I remember, it was added to communicate error details when the communication channel fails, in a way that might help applications recover if they reconnect in the future.

See the notes in https://w3c.github.io/presentation-api/#sending-a-message-through-presentationconnection

In retrospect, this probably isn't and won't get used for anything in the real world, and removing closeMessage would be a reasonable spec improvement.